File: data/imported/knowledge/email/net/merged/snippets/snippet_003.cs
[Fact]
    public void WriterRejectsDuplicateSiblingNamesUnderCfbOrdering()
    {
        var document = new CfbDocument();
        document.Root.AddStream(new CfbStream("ab", [1]));
        document.Root.AddStream(new CfbStream("AB", [2]));

        var error = Assert.Throws<CfbException>(() => CfbWriter.ToBytes(document));
        Assert.Contains("Duplicate sibling name", error.Message);
    }

File: data/imported/knowledge/email/python/merged/snippets/snippet_003.py
# Adapted from aspose.org: knowledge/email/python/merged/snippets/snippet_003.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_large_v3_document_emits_difat(self) -> None:

        payload = b"Z" * (512 * 15000)



        root = CFBStorage(ROOT_ENTRY_NAME)

        root.add_stream(CFBStream("Huge", payload))



        reader = CFBReader(CFBWriter.to_bytes(CFBDocument(root=root, major_version=3)))



        self.assertGreater(reader.header.number_of_fat_sectors, 109)

        self.assertGreater(reader.header.number_of_difat_sectors, 0)

        stream_entry = reader.resolve_path(["Huge"])

        self.assertIsNotNone(stream_entry)

        assert stream_entry is not None

        self.assertEqual(reader.get_stream_data(stream_entry.stream_id), payload)